### PR TITLE
Fix some errors

### PR DESCRIPTION
**Pull request by kosssi**
- fix isDispatched when doc not exist
- update comments
- fix appExist typo
- fix replication launcher when is already started

### CODE REVIEW

File: www-src/app/replicator/fromDevice/event_synchronizer.coffee [hunk 1]
```diff
@@ -95,7 +97,12 @@ module.exports = class EventSynchronizer
             _deleted: true
 
         @db.put toDelete, toDelete._id, toDelete._rev, (err, res) =>
-            return callback err if err
+            if err
+                if err.status is 409
+                    @conflictsHandler.handleConflicts toDelete, (err, doc) =>
+                        return callback err if err
```
Comment (nono) [inline www-src/app/replicator/fromDevice/event_synchronizer.coffee]: It looks like the workflow is a bit broken here. `@conflictsHandler.handleConflicts` and `@calendarSync.deleteEvent` run concurrently. And if `handleConflicts` returns an `err`, `callback` is called twice.

**Comment by kosssi** [comment]
thanks @nono

File: www-src/app/replicator/conflicts_handler.coffee [hunk 2]
```diff
@@ -11,7 +11,9 @@ module.exports = class ConflictsHandler
     handleConflicts: (doc, callback) ->
         # Get the doc with conflicts (and revs) infos from Pouch
         @db.get doc._id, { conflicts: true, revs: true }, (err, local) =>
-            return callback err if err
+            if err
+                return callback null, doc if err.status is 404
```
Comment (jacquarg) [inline www-src/app/replicator/conflicts_handler.coffee]: If we get a 404 here, it means that we make this work too soon (ie before PouchDB store the changes)

File: www-src/app/replicator/fromDevice/event_synchronizer.coffee [hunk 3]
```diff
@@ -95,7 +97,15 @@ module.exports = class EventSynchronizer
             _deleted: true
 
         @db.put toDelete, toDelete._id, toDelete._rev, (err, res) =>
-            return callback err if err
-
-            @calendarSync.deleteEvent androidEvent, \
-                    AndroidAccount.ACCOUNT, callback
+            if err
+                if err.status is 409
+                    @conflictsHandler.handleConflicts toDelete, (err, doc) =>
```
Comment (jacquarg) [inline www-src/app/replicator/fromDevice/event_synchronizer.coffee]: Not sure it's the right conflit you're trying to solve here : 409 are for "immediate conflicts" (trying to update a doc from a to older version ; but conflictHandler work on eventual conflict ( http://pouchdb.com/guides/conflicts.html )
But right, we have to handle immediate conflict in import code !

File: www-src/app/replicator/main.coffee [hunk 4]
```diff
@@ -638,7 +638,9 @@ module.exports = class Replicator extends Backbone.Model
 
         ReplicationLauncher = require "./replication_launcher"
         @replicationLauncher = new ReplicationLauncher @config, app.router
-        @replicationLauncher.start options, callback
+        @replicationLauncher.start options, =>
+            @stopRealtime()
```
Comment (jacquarg) [inline www-src/app/replicator/main.coffee]: Ok, that's not obvious why you are calling stopRealtime here.
It's just to clean up the @replicationLauncher objects ?

File: www-src/app/replicator/replication_launcher.coffee [hunk 5]
```diff
@@ -95,13 +95,13 @@ module.exports = class ReplicationLauncher
             liveOptions =
               retry: true
               back_off_function: (delay) ->
+                  console.log "back_off_function", delay
```
Comment (jacquarg) [inline www-src/app/replicator/replication_launcher.coffee]: console.log  !

File: www-src/app/replicator/change/conflicts_handler.coffee [hunk 6]
```diff
@@ -9,8 +9,11 @@ module.exports = class ConflictsHandler
     constructor: (@db)->
 
     handleConflicts: (doc, callback) ->
+        log.info "handleConflicts"
+
         # Get the doc with conflicts (and revs) infos from Pouch
-        @db.get doc._id, { conflicts: true, revs: true }, (err, local) =>
+        @db.get doc._id, { conflicts: true, revs: true, open_revs: all }, \
```
Comment (jacquarg) [inline www-src/app/replicator/change/conflicts_handler.coffee]: where is all defined ?

Comment (nono) [inline www-src/app/replicator/change/conflicts_handler.coffee]: +1 for `'all'`

**Comment by kosssi** [comment]
@jacquarg @nono is better thanks

**Comment by jacquarg** [comment]
: +1 : 

We are actually testing the APK of this branch

**Pull request merged**